crypto/err/err.c

```diff
@@ -155,22 +155,19 @@ static int err_string_data_cmp(const ERR_STRING_DATA *a,
     return (int)(a->error - b->error);
 }
 
-static LHASH_OF(ERR_STRING_DATA) *get_hash(int create, int lockit)
+/*
+ * The err_string_lock must be held by the caller.  Either read or write
+ * lock is acceptable.
+ */
+static LHASH_OF(ERR_STRING_DATA) *get_hash(int create)
 {
     LHASH_OF(ERR_STRING_DATA) *ret = NULL;
 
-    if (lockit)
-        CRYPTO_THREAD_write_lock(err_string_lock);
     if (!int_error_hash && create) {
         int_error_hash = lh_ERR_STRING_DATA_new(err_string_data_hash,
                                                 err_string_data_cmp);
     }
-    if (int_error_hash != NULL)
-        ret = int_error_hash;
-    if (lockit)
-        CRYPTO_THREAD_unlock(err_string_lock);
-
-    return ret;
+    return int_error_hash;
```

Comment:
If `create` is 0, it seems this function just amounts to `return int_error_hash`. Dunno if you'd rather just inline it.